src/htmx.js

```diff
@@ -3523,7 +3523,7 @@ var htmx = (function() {
 
     // for a non-GET include the closest form
     if (verb !== 'get') {
-      processInputValue(processed, priorityFormData, errors, closest(elt, 'form'), validate)
```

Comment:
would you mind extracting this to a local variable to avoid the complicated expression in the parameter position, w/ maybe a comment explaining the situation?

thank you, PR looks great love the tests!

Reply:
I addressed your request, but also there's been some discussion about a better way forward on #issues-and-pull-requests that I'm happy to implement. 

One thing that might need clarifying is the original intent of the sentence in the docs that this PR changes. Specifically, what kind of elements inside a form should also be sending form values on non-GET?

- `<select hx-post=/foo>`?
- `<input type=text hx-post=/foo>`? 
- `<input type=submit hx-post=/foo>`?
- `<div hx-post=/foo>`?

Furthermore, personally I'm wondering why this is being limited to non-GET? Wouldn't we want to submit the entire form for e.g.

```
<form method=get action=/foo>
  <input name=bar value=baz>
  <button hx-get=/foo>Submit</button>
</form>
```

I'm actually not 100% on whether HTMX currently submits `bar=baz` in this situation or not (I haven't been able to find an associated test and I'm on mobile now), but the docs and the code appear to indicate that it won't. I will make sure to cover any of those situations with tests once a decision has been made.

In case it's helpful, here's the original PR of that line in the docs: https://github.com/bigskysoftware/htmx/commit/a3c9cf6b6a3fe05b8207e703bd6ff16fd51e34f2#diff-c7d06af83d22e155470d44a366cc1ba76cd5dc9da3405fc9e54a237174736b6aR962

Reply:
htmx has not submitted values for `GET`s that occur within a form so that link-like things do not include them when navigating.  This behavior may not be ideal but we can't change it without breaking backwards compatibility.  (In general at this point backwards compatibility is becoming more important than correctness in htmx.)

I think the situation where the user has explicitly called out a form via the form attribute is obvious enough to fix so I'm in favor of this change so long as it minimizes any additional changes, in particular with an eye towards backwards compatibility.